crates/sui-core/src/authority/package_object_cache.rs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use parking_lot::RwLock;
+use std::collections::HashMap;
+use std::sync::Arc;
+use sui_types::base_types::ObjectID;
+use sui_types::error::{SuiError, SuiResult, UserInputError};
+use sui_types::object::Object;
+use sui_types::storage::{BackingPackageStore, ObjectStore};
+
+pub struct PackageObjectCache<S> {
+    cache: RwLock<HashMap<ObjectID, Object>>,
```

Comment:
we should probably use `LruCache` (which we use elsewhere) to keep the size of this bounded.